pkg/api/constants/extensions.go

```diff
@@ -5,8 +5,10 @@ const (
 	ExtCatalogPrefix     = "/_catalog"
 	ExtOciDiscoverPrefix = "/_oci/ext/discover"
 	// zot specific extensions.
-	ExtSearchPrefix  = "/_zot/ext/search"
-	FullSearchPrefix = RoutePrefix + ExtSearchPrefix
-	ExtMgmtPrefix    = "/_zot/ext/mgmt"
-	FullMgmtPrefix   = RoutePrefix + ExtMgmtPrefix
+	ExtSearchPrefix       = "/_zot/ext/search"
+	FullSearchPrefix      = RoutePrefix + ExtSearchPrefix
+	ExtMgmtPrefix         = "/_zot/ext/mgmt"
+	FullMgmtPrefix        = RoutePrefix + ExtMgmtPrefix
+	ExtPreferencesPrefix  = "/_zot/ext/user/preferences"
```

Comment:
/_zot/ext/userprefs

Reply:
alternative: /_zot/ext/mgmt/userprefs

Reply:
https://github.com/opencontainers/distribution-spec/tree/main/extensions#name

Reply:
_<extension>/<component>/<module>[?<key>=<value>&...]

Reply:
_extension = _zot
component = ext
module = search

Reply:
_zot/registry/search?

Reply:
_zot/ext/userprefs?repo=REPO&bookmark=true&star=false

Reply:
Since setting a bookmark or a star are separate operations we came up with this request format:
`_zot/ext/userprefs?action=toggleStar&repo=repo`

Currently we have the actions "toggleStar" and "toggleBookmark" that require the extra parameter "repo".

If we'll want to add more actions with different parameters it will be pretty easy to do.